arangod/RestServer/QueryRegistryFeature.cpp

```diff
@@ -412,23 +415,6 @@ void QueryRegistryFeature::validateOptions(
     FATAL_ERROR_EXIT();
   }
 
-  if (_queryMaxRuntime < 0.0) {
-    LOG_TOPIC("46572", FATAL, Logger::AQL)
-        << "invalid value for `--query.max-runtime`. expecting 0 or a positive "
-           "value";
-    FATAL_ERROR_EXIT();
-  }
-
-  if (_maxQueryPlans == 0) {
-    LOG_TOPIC("4006f", FATAL, Logger::AQL)
-        << "invalid value for `--query.optimizer-max-plans`. expecting at "
-           "least 1";
-    FATAL_ERROR_EXIT();
-  }
-
-  // cap the value somehow. creating this many plans really does not make sense
-  _maxQueryPlans = std::min(_maxQueryPlans, decltype(_maxQueryPlans)(1024));
```

Comment:
This changes the behaviour, since this limit does not seem to be enforced any more (the maximum is UINT64_MAX).

Reply:
Good catch. That part was unintentionally deleted. Fixed now!